File: lib/src/main/java/com/ticketmaster/structurizr/threagile/ThreagileWorkspaceExporter.java
package com.ticketmaster.structurizr.threagile;

import com.structurizr.Workspace;
import com.structurizr.export.AbstractWorkspaceExporter;
import com.structurizr.export.WorkspaceExport;

public class ThreagileWorkspaceExporter extends AbstractWorkspaceExporter {
    public WorkspaceExport export(Workspace workspace) {
        return new ThreagileWorkspaceExport("Some yaml");
    }
}

File: lib/src/main/kotlin/com/ticketmaster/structurizr/ThreagileWorkspaceExporter.kt
package com.ticketmaster.structurizr

import com.structurizr.Workspace
import com.structurizr.export.WorkspaceExport
import com.structurizr.export.WorkspaceExporter


open class ThreagileWorkspaceExporter : WorkspaceExporter {
    override fun export(workspace: Workspace?): WorkspaceExport {
        return try {
            YamlWorkspaceExport("Some yaml")
        } catch (e: Exception) {
            throw RuntimeException(e)
        }
    }
}